persistence/src/main/java/edu/unc/lib/dl/persist/services/destroy/DestroyObjectsJob.java

```diff
@@ -0,0 +1,154 @@
+import edu.unc.lib.dl.fcrepo4.ContentObject;
+import edu.unc.lib.dl.fcrepo4.FedoraTransaction;
+import edu.unc.lib.dl.fcrepo4.FileObject;
+import edu.unc.lib.dl.fcrepo4.RepositoryObject;
+import edu.unc.lib.dl.fcrepo4.RepositoryObjectFactory;
+import edu.unc.lib.dl.fcrepo4.RepositoryObjectLoader;
+import edu.unc.lib.dl.fcrepo4.TransactionManager;
+import edu.unc.lib.dl.fedora.FedoraException;
+import edu.unc.lib.dl.fedora.PID;
+import edu.unc.lib.dl.metrics.TimerFactory;
+import edu.unc.lib.dl.rdf.Cdr;
+import edu.unc.lib.dl.rdf.DcElements;
+import edu.unc.lib.dl.rdf.Ebucore;
+import edu.unc.lib.dl.rdf.Premis;
+import edu.unc.lib.dl.search.solr.service.ObjectPathFactory;
+import io.dropwizard.metrics5.Timer;
+
+/**
+ *
+ * @author harring
+ *
+ */
+public class DestroyObjectsJob implements Runnable {
+    private static final Timer timer = TimerFactory.createTimerForClass(DestroyObjectsJob.class);
+
+    private List<PID> objsToDestroy;
+    private AgentPrincipals agent;
+    private RepositoryObjectFactory repoObjFactory;
+    private RepositoryObjectLoader repoObjLoader;
+    private TransactionManager txManager;
+    private DestroyProxyService proxyService;
+    private ObjectPathFactory pathFactory;
+
+    public DestroyObjectsJob(AgentPrincipals agent, List<PID> objsToDestroy) {
+        this.objsToDestroy = objsToDestroy;
+    }
+
+    @Override
+    public void run() {
+        FedoraTransaction tx = txManager.startTransaction();
+        try (Timer.Context context = timer.time()) {
+            // convert each destroyed obj to a tombstone
+            for (PID pid : objsToDestroy) {
+                // remove containment relation from obj's parent
+                proxyService.destroyProxy(pid);
+                RepositoryObject repoObj = repoObjLoader.getRepositoryObject(pid);
+                // purge tree with repoObj as root from repository
+                destroyTree(repoObj);
+           }
+        } catch (Exception e) {
+             tx.cancel(e);
+        } finally {
+             tx.close();
+        }
+    }
+
+    private void destroyTree(RepositoryObject rootOfTree) throws FedoraException {
+        if (rootOfTree instanceof ContentContainerObject) {
+            ContentContainerObject container = (ContentContainerObject) rootOfTree;
+            List<ContentObject> members = container.getMembers();
+            for (ContentObject member : members) {
+                destroyTree(member);
+            }
+        } else if (rootOfTree instanceof FileObject) {
+            FileObject file = (FileObject) rootOfTree;
+            List<BinaryObject> binaries = file.getBinaryObjects();
+            for (BinaryObject binary : binaries) {
+                destroyTree(binary);
+            }
+        }
+        // destroy root of tree
+        Model stoneModel = rootOfTree.getModel();
+        stoneModel = convertModelToTombstone(rootOfTree);
+        repoObjFactory.createObject(rootOfTree.getUri(), stoneModel);
```

Comment:
Its kind of weirding me out that the method is named createObject when its being used to PUT to an existing resource, but it looks like it should work